testsuite/testclient/conftest.py

```diff
@@ -0,0 +1,900 @@
+"""
+Module for running the testclient .yaml test cases.
```

Comment:
Can we document how this all comes together between this and the yaml files.  We know what is happening but not sure anybody else would without a lot of work.  I.e this is diriven by the yaml files in dir testclient, etc.

QUESTION: Where did the name conftest in conftest.py come from???

Reply:
I have struggled more that normal trying to sort out how this all ties togther but part of that is that it is significantly different from the original code.

Reply:
Will add an explanation. conftest.py is a hard coded name that pytest recognizes.

Reply:
DONE. Documented the test approach, and the testcase YAML syntax.